service/frontend/workflow_handler.go

```
@@ -2092,69 +2092,90 @@ func (wh *WorkflowHandler) ListOpenWorkflowExecutions(ctx context.Context, reque
+			enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING,
+		),
+	}
+
 	if request.StartTimeFilter == nil {
 		request.StartTimeFilter = &filterpb.StartTimeFilter{}
 	}
 
 	if request.StartTimeFilter.EarliestTime == nil || request.StartTimeFilter.EarliestTime.AsTime().IsZero() {
 		request.StartTimeFilter.EarliestTime = timestamppb.New(minTime)
+	} else {
+		baseReq.Query += fmt.Sprintf(
+			"AND %s >= '%s'",
+			searchattribute.StartTime,
+			request.StartTimeFilter.EarliestTime.AsTime().Format(time.RFC3339Nano),
+		)
 	}
 
 	if request.StartTimeFilter.LatestTime == nil || request.StartTimeFilter.LatestTime.AsTime().IsZero() {
 		request.StartTimeFilter.LatestTime = timestamppb.New(maxTime)
+	} else {
+		baseReq.Query += fmt.Sprintf(
+			"AND %s <= '%s'",
+			searchattribute.StartTime,
+			request.StartTimeFilter.LatestTime.AsTime().Format(time.RFC3339Nano),
+		)
 	}
 
 	if request.StartTimeFilter.EarliestTime.AsTime().After(request.StartTimeFilter.LatestTime.AsTime()) {
 		return nil, errEarliestTimeIsGreaterThanLatestTime
 	}
 
-	maxPageSize := int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
-	if request.GetMaximumPageSize() <= 0 || request.GetMaximumPageSize() > maxPageSize {
-		request.MaximumPageSize = maxPageSize
-	}
-
-	namespaceName := namespace.Name(request.GetNamespace())
-	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
-	if err != nil {
-		return nil, err
-	}
-
-	baseReq := &manager.ListWorkflowExecutionsRequest{
-		NamespaceID:       namespaceID,
-		Namespace:         namespaceName,
-		PageSize:          int(request.GetMaximumPageSize()),
-		NextPageToken:     request.NextPageToken,
-		EarliestStartTime: request.StartTimeFilter.EarliestTime.AsTime(),
-		LatestStartTime:   request.StartTimeFilter.LatestTime.AsTime(),
-	}
-
 	var persistenceResp *manager.ListWorkflowExecutionsResponse
 	if request.GetExecutionFilter() != nil {
 		if wh.config.DisableListVisibilityByFilter(namespaceName.String()) {
 			err = errListNotAllowed
```

This looks like it should return an error instead of setting the error.